cmd/proxy/mocks_test.go

```diff
@@ -0,0 +1,253 @@
+	}
+
+	msg := &dns.Msg{}
+	msg.SetQuestion(dns.Fqdn(name), qt)
+	msg.RecursionDesired = recursion
+	msg.SetEdns0(4096, dnssec)
+
+	wire, err := msg.Pack()
+	if err != nil {
+		panic(fmt.Errorf("pack query: %w", err))
+	}
+
+	return &MockQUICStream{
+		ID:     quic.StreamID(id),
+		InBuf:  wire,
+		OutBuf: make([]byte, 4096),
+
+		msg: msg,
+	}
+}
+
+func (mqs *MockQUICStream) GetMsg() *dns.Msg {
+	return mqs.msg
+}
+
+// ************************************
+// quic.ReceiveStream interface methods
+// ************************************
+
+// StreamID returns the stream ID.
+func (mqs *MockQUICStream) StreamID() quic.StreamID {
+	return mqs.ID
+}
+
+// Read reads data from the stream.
+// Read can be made to time out and return a net.Error with Timeout() == true
+// after a fixed time limit; see SetDeadline and SetReadDeadline.
+// If the stream was canceled by the peer, the error implements the StreamError
+// interface, and Canceled() == true.
+// If the session was closed due to a timeout, the error satisfies
+// the net.Error interface, and Timeout() will be true.
+func (mqs *MockQUICStream) Read(buf []byte) (int, error) {
+	if mqs.InBuf == nil {
+		return 0, io.EOF
+	}
+
+	return copy(buf, mqs.InBuf), io.EOF
+}
+
+// CancelRead aborts receiving on this stream.
+// It will ask the peer to stop transmitting stream data.
+// Read will unblock immediately, and future Read calls will fail.
+// When called multiple times or after reading the io.EOF it is a no-op.
+func (mqs *MockQUICStream) CancelRead(quic.ErrorCode) {}
+
+func (mqs *MockQUICStream) SetReadDeadline(t time.Time) error {
+	return ErrUnimplementedMock
+}
+
+// *********************************
+// quic.SendStream interface methods
+// *********************************
+
+// Write writes data to the stream.
+// Write can be made to time out and return a net.Error with Timeout() == true
+// after a fixed time limit; see SetDeadline and SetWriteDeadline.
+// If the stream was canceled by the peer, the error implements the StreamError
+// interface, and Canceled() == true.
+// If the session was closed due to a timeout, the error satisfies
+// the net.Error interface, and Timeout() will be true.
+func (mqs *MockQUICStream) Write(p []byte) (int, error) {
+	mqs.OutBuf = make([]byte, len(p))
+
+	return copy(mqs.OutBuf, p), nil
+}
+
+// Close closes the write-direction of the stream.
+// Future calls to Write are not permitted after calling Close.
+// It must not be called concurrently with Write.
+// It must not be called after calling CancelWrite.
+func (mqs *MockQUICStream) Close() error {
+	return nil
+}
+
+// The context is canceled as soon as the write-side of the stream is closed.
+// This happens when Close() or CancelWrite() is called, or when the peer
+// cancels the read-side of their stream.
+// Warning: This API should not be considered stable and might change soon.
+func (mqs *MockQUICStream) Context() context.Context {
+	panic(ErrUnimplementedMock)
```

Comment:
Please, can you make the error'ing behavior for unimplemented methods consistent? I suggest `panic("not implemented")`. In a few cases above you return the error from the function or just nil pointer if there is no error in the return signature.